tests/SecretsCrypto/tst_secretscrypto/tst_secretscrypto.c

```diff
@@ -1567,6 +1569,449 @@ static void tst_crypto_import_invalid(SfCryptoFixture *fixture, gconstpointer te
+            NULL,
+            CRYPTO_PLUGIN_TEST,
+            NULL,
+            _tst_crypto_ref_res_and_quit,
+            fixture);
+    g_main_loop_run(fixture->loop);
+
+    verifier = sf_crypto_cipher_session_new_finish(fixture->test_res, &fixture->error);
+
+    g_assert_no_error(fixture->error);
+    g_assert_nonnull(verifier);
+
+    if (g_test_failed())
+        return;
+
+    g_test_queue_unref(verifier);
+
+    for (i = 0; i < kd->chunks; i++) {
+        GBytes *random_bytes;
+
+        sf_crypto_manager_generate_random_data(fixture->manager,
+                kd->data_size,
+                SF_CRYPTO_DEFAULT_CSPRNG_ENGINE,
+                NULL,
+                CRYPTO_PLUGIN_TEST,
+                NULL,
+                _tst_crypto_ref_res_and_quit,
+                fixture);
+        g_main_loop_run(fixture->loop);
+
+        random_bytes = sf_crypto_manager_generate_random_data_finish(fixture->test_res, &fixture->error);
+
+        g_assert_no_error(fixture->error);
+        g_assert_nonnull(random_bytes);
+
+        if (g_test_failed())
+            return;
+
+        sf_crypto_cipher_session_update(signer,
+                random_bytes,
+                NULL,
+                NULL,
+                _tst_crypto_ref_res_and_quit,
+                fixture);
+        g_main_loop_run(fixture->loop);
+
+        sf_crypto_cipher_session_update_finish(fixture->test_res, NULL, &fixture->error);
+
+        g_assert_no_error(fixture->error);
+
+        if (g_test_failed())
+            return;
+
+        sf_crypto_cipher_session_update(verifier,
+                random_bytes,
+                NULL,
+                NULL,
+                _tst_crypto_ref_res_and_quit,
+                fixture);
+        g_bytes_unref(random_bytes);
+        g_main_loop_run(fixture->loop);
+
+        sf_crypto_cipher_session_update_finish(fixture->test_res, NULL, &fixture->error);
+
+        g_assert_no_error(fixture->error);
+
+        if (g_test_failed())
+            return;
+    }
+
+    sf_crypto_cipher_session_close(signer,
+            NULL,
+            NULL,
+            NULL,
+            _tst_crypto_ref_res_and_quit,
+            fixture);
+    g_main_loop_run(fixture->loop);
+
+    sf_crypto_cipher_session_close_finish(fixture->test_res, &signature, NULL, &fixture->error);
+
+    g_assert_no_error(fixture->error);
+    g_assert_nonnull(signature);
+
+    if (g_test_failed())
+        return;
+
+    sf_crypto_cipher_session_close(verifier,
+            signature,
+            NULL,
+            NULL,
+            _tst_crypto_ref_res_and_quit,
+            fixture);
+    g_main_loop_run(fixture->loop);
+    g_bytes_unref(signature);
+
+    sf_crypto_cipher_session_close_finish(fixture->test_res,
+            NULL,
+            &verify_status,
+            &fixture->error);
+
+    g_assert_no_error(fixture->error);
+
+    if (g_test_failed())
+        return;
+
+    g_assert_cmpint(verify_status, ==, SF_CRYPTO_VERIFICATION_STATUS_SUCCEEDED);
```

Comment:
It might also be worthwhile to ensure that if you deliberately mess with random_bytes just prior to feeding them into the verifier, that verification fails, just to ensure that the API isn't just returning VERIFICATION_STATUS_SUCCEEDED for every input, etc.

Reply:
i.e. in a separate / second "expect failure" test.